### ISSUE TITLE
Every unit created by runc need daemon reload since systemd v230.

### ISSUE THREAD

**Issue by wenjianhn** [opencontainers/runc#4568]
### Description

Every unit created by runc need daemon reload since systemd v230.

### Steps to reproduce the issue

1.  sudo  runc --systemd-cgroup --root /tmp/runc run mycontainerid
2.  sudo systemctl show -P NeedDaemonReload runc-mycontainerid.scope

### Describe the results you received and expected

Actual:
```
$ sudo systemctl show -P NeedDaemonReload runc-mycontainerid.scope
yes
```

Expected:
```
$ sudo systemctl show -P NeedDaemonReload runc-mycontainerid.scope
no
```

### What version of runc are you using?

v1.2.0-109

**Comment by kolyshkin** [comment]
So, does runc properly generates `DeviceAllow` entries for nvidia devices (the subject of #3708). If no, what is the reason? If yes, why does systemd restart breaks it?

**Comment by wenjianhn** [comment]
@kolyshkin No. The `DeviceAllow` entries doesn't have any nvidia devices since we have not enabled  [this workaround](https://github.com/NVIDIA/nvidia-docker/issues/1671#issuecomment-1330466432).

A runc hook was used to inject the devices to the cgroup(device.allow) behind the back of systemd.
The injection will be flushed away by `systemctl daemon-reload`.

See https://github.com/NVIDIA/nvidia-docker/issues/1671#issuecomment-1330894367

**Comment by wenjianhn** [comment]
I didn't know that `systemctl daemon-reload` would reload all the units(even if configuration had not changed on disk).
Reloading all the units in order to make the configuration of a single unit up to date :(

As a matter of fact, this issue doesn't break support for NVIDIA GPUs.
cc @vsxen @cyphar

**Comment by kolyshkin** [comment]
> @kolyshkin No. The `DeviceAllow` entries doesn't have any nvidia devices since we have not enabled [this workaround](https://github.com/NVIDIA/nvidia-docker/issues/1671#issuecomment-1330466432).

There is no need for any workarounds or hacks -- if you use runc >= 1.1.7 and systemd >= v240, it should work with the stock runc (see https://github.com/opencontainers/runc/issues/3671#issuecomment-1533338473). This will also solve the issue of removing the access to nvidia devices upon daemon-reload, as systemd is now aware of all the devices.
